src/m365/spo/commands/applicationcustomizer/applicationcustomizer-get.ts

```diff
@@ -0,0 +1,177 @@
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId  [clientSideComponentId]'
+      },
+      {
+        option: '-s, --scope [scope]'
+      },
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const customAction = await this.getCustomAction(args.options);
+
+      if (customAction) {
+        logger.log({
+          ClientSideComponentId: customAction.ClientSideComponentId,
+          ClientSideComponentProperties: customAction.ClientSideComponentProperties,
+          CommandUIExtension: customAction.CommandUIExtension,
+          Description: customAction.Description,
+          Group: customAction.Group,
+          Id: customAction.Id,
+          ImageUrl: customAction.ImageUrl,
+          Location: customAction.Location,
+          Name: customAction.Name,
+          RegistrationId: customAction.RegistrationId,
+          RegistrationType: customAction.RegistrationType,
+          Rights: JSON.stringify(customAction.Rights),
+          Scope: this.humanizeScope(customAction.Scope),
+          ScriptBlock: customAction.ScriptBlock,
+          ScriptSrc: customAction.ScriptSrc,
+          Sequence: customAction.Sequence,
+          Title: customAction.Title,
+          Url: customAction.Url,
+          VersionOfUserCustomAction: customAction.VersionOfUserCustomAction
+        });
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedPromise(err);
+    }
+  }
+
+  private async getCustomAction(options: Options): Promise<CustomAction | undefined> {
+    const scope = options.scope ? options.scope : 'All';
+
+    if (options.id) {
+      const customAction = await spo.getCustomActionById(options.webUrl, options.id, scope);
+
+      if (!customAction) {
+        throw `No application customizer with id '${options.id}' found`;
+      }
+
+      if (customAction.Location !== 'ClientSideExtension.ApplicationCustomizer') {
+        throw 'The found custom action is not an Application Customizer';
+      }
```

Comment:
this may be a bit confiusing for the user that she or he was looking for application customizer but will see some error message that in fact the custom action found is not an application customizer 🤔. I think the idea behind this command is to filter and consider only application customizers and do not care about extensions.